clients/src/test/java/org/apache/kafka/common/network/SslSelectorTest.java

```diff
@@ -160,6 +165,85 @@ public void testDisabledRenegotiation() throws Exception {
 
     }
 
+    @Override
+    public void testMuteOnOOM() throws Exception {
+        //clean up default selector, replace it with one that uses a finite mem pool
+        selector.close();
+        MemoryPool pool = new SimpleMemoryPool(900, 900, false);
+        //the initial channel builder is for clients, we need a server one
+        File trustStoreFile = File.createTempFile("truststore", ".jks");
+        Map<String, Object> sslServerConfigs = TestSslUtils.createSslConfig(false, true, Mode.SERVER, trustStoreFile, "server");
+        sslServerConfigs.put(SslConfigs.PRINCIPAL_BUILDER_CLASS_CONFIG, Class.forName(SslConfigs.DEFAULT_PRINCIPAL_BUILDER_CLASS));
+        channelBuilder = new SslChannelBuilder(Mode.SERVER);
+        channelBuilder.configure(sslServerConfigs);
+        selector = new Selector(NetworkReceive.UNLIMITED, 5000, metrics, time, "MetricGroup", 
+                new HashMap<String, String>(), true, false, channelBuilder, pool);
+
+        try (ServerSocketChannel ss = ServerSocketChannel.open()) {
+            ss.bind(new InetSocketAddress(0));
+
+            InetSocketAddress serverAddress = (InetSocketAddress) ss.getLocalAddress();
+
+            SslSender sender1 = createSender(serverAddress, randomPayload(900));
+            SslSender sender2 = createSender(serverAddress, randomPayload(900));
+            sender1.start();
+            sender2.start();
+
+            SocketChannel channelX = ss.accept(); //not defined if its 1 or 2
+            channelX.configureBlocking(false);
+            SocketChannel channelY = ss.accept();
+            channelY.configureBlocking(false);
+            selector.register("clientX", channelX);
+            selector.register("clientY", channelY);
+
+            boolean success = false;
+            NetworkReceive firstReceive = null;
+            long deadline = System.currentTimeMillis() + 5000;
+            //keep calling poll until:
+            //1. both senders have completed the handshakes (so server selector has tried reading both payloads)
+            //2. a single payload is actually read out completely (the other is too big to fit)
+            while (System.currentTimeMillis() < deadline) {
+                selector.poll(10);
+
+                List<NetworkReceive> completed = selector.completedReceives();
+                if (firstReceive == null) {
+                    if (!completed.isEmpty()) {
+                        assertEquals("expecting a single request", 1, completed.size());
+                        firstReceive = completed.get(0);
+                        assertTrue(selector.isMadeProgressLastPoll());
+                        assertEquals(0, pool.availableMemory());
+                    }
+                } else {
+                    assertTrue("only expecting single request", completed.isEmpty());
+                }
+
+                boolean handshaked = sender1.waitForHandshake(1);
+                handshaked &= sender2.waitForHandshake(1);
```

Comment:
this assignment operator is defined differently for boolean operands (so it doesnt perform a bitwise operation) and so there is no &&=. i'll refactor the code to avoid this (as its rather obscure)